modules/consensus/synchronize_test.go

```diff
@@ -225,3 +225,69 @@ func TestBlockHistory(t *testing.T) {
 		}
 	}
 }
+
+// TestSendBlocksBroadcasts tests that the SendBlocks RPC call only Broadcasts
+// one block, no matter how many blocks are sent. In the case 0 blocks are
+// sent, tests that Broadcast is never called.
+func TestSendBlocksBroadcasts(t *testing.T) {
+	// Setup consensus sets.
+	cst1, err := blankConsensusSetTester("TestSendBlocksBroadcastsOnce1")
+	if err != nil {
+		t.Fatal(err)
+	}
+	defer cst1.Close()
+	cst2, err := blankConsensusSetTester("TestSendBlocksBroadcastsOnce2")
+	if err != nil {
+		t.Fatal(err)
+	}
+	defer cst2.Close()
+	// Setup mock gateway.
+	mg := mockGateway{Gateway: cst1.cs.gateway}
+	cst1.cs.gateway = &mg
+	err = cst1.cs.gateway.Connect(cst2.cs.gateway.Address())
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	tests := []struct {
+		blocksToMine          int
+		expectedNumBroadcasts int
+	}{
+		{0, 0},
+		{1, 1},
+		{2, 1},
+		{MaxCatchUpBlocks, 1},
+		{2 * MaxCatchUpBlocks, 1},
```

Comment:
also, shouldn't `go vet` be complaining about the way you are initializing the struct? You are using unnamed fields.